lab_1_keywords_tfidf/main.py

```diff
@@ -99,7 +139,18 @@ def calculate_tfidf(term_freq: dict[str, float], idf: dict[str, float]) -> Optio
 
     In case of corrupt input arguments, None is returned
     """
-    pass
+    if isinstance(term_freq, dict) and all(isinstance(token, str) for token in term_freq.keys()) and all(
+            isinstance(value, float) for value in term_freq.values()) and isinstance(idf, dict) and all(
+            isinstance(token, str) for token in idf.keys()) and all(isinstance(value, float) for value in idf.values()):
+        tfidf_dict = {}
+        for token in term_freq.keys():
+            if token in idf.keys():
+                tfidf_dict[token] = term_freq.get(token) * idf.get(token)
+            else:
+                tfidf_dict[token] = term_freq.get(token) * log(47 / (0 + 1))
```

Comment:
why do you do  0+ 1?

Reply:
that's how it was in the formula! but I will replace it